shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/hint/HintManagerTest.java

```diff
@@ -111,6 +111,32 @@ public void assertIsDatabaseShardingOnlyWithoutSet() {
         assertFalse(HintManager.isDatabaseShardingOnly());
     }
     
+    @Test
+    public void assertAddDatabaseShardingValueOnlyDatabaseSharding() {
+        try (HintManager hintManager = HintManager.getInstance()) {
+            hintManager.setDatabaseShardingValue(1);
+            assertTrue(HintManager.isDatabaseShardingOnly());
+            hintManager.addDatabaseShardingValue("logic_table", 2);
+            assertFalse(HintManager.isDatabaseShardingOnly());
+            assertThat(HintManager.getDatabaseShardingValues("logic_table").size(), is(1));
+            assertTrue(HintManager.getDatabaseShardingValues("logic_table").contains(2));
+            hintManager.close();
+        }
+    }
+    
+    @Test
+    public void assertAddTableShardingValueOnlyDatabaseSharding() {
+        try (HintManager hintManager = HintManager.getInstance()) {
+            hintManager.setDatabaseShardingValue(1);
+            assertTrue(HintManager.isDatabaseShardingOnly());
+            hintManager.addTableShardingValue("logic_table", 2);
+            assertFalse(HintManager.isDatabaseShardingOnly());
+            assertThat(HintManager.getTableShardingValues("logic_table").size(), is(1));
+            assertTrue(HintManager.getTableShardingValues("logic_table").contains(2));
+            hintManager.close();
```

Comment:
Is it necessary to call `close` again?
Try with resources is close the resource.